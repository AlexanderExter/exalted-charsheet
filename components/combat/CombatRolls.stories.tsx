import React, { useState } from "react"
import { CombatRolls } from "./CombatRolls"
import { basicCharacter } from "./combatStories.helpers"
import type { Character } from "@/lib/character-types"

const meta = { title: "Components/CombatRolls", component: CombatRolls }
export default meta

export const Basic = () => {
  const [character, setCharacter] = useState<Character>(basicCharacter)

  return (
    <CombatRolls
      character={character}
      updateCharacter={updates => setCharacter(prev => ({ ...prev, ...updates }))}
      getHighestAttribute={() => 3}
    />
  )
}

