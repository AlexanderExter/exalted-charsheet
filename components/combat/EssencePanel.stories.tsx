import React, { useState } from "react"
import { EssencePanel } from "./EssencePanel"
import { basicCharacter } from "./combatStories.helpers"
import type { Character } from "@/lib/character-types"

const meta = { title: "Components/EssencePanel", component: EssencePanel }
export default meta

export const Basic = () => {
  const [character, setCharacter] = useState<Character>(basicCharacter)

  return (
    <EssencePanel
      character={character}
      updateCharacter={updates => setCharacter(prev => ({ ...prev, ...updates }))}
    />
  )
}

