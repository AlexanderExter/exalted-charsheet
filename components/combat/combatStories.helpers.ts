import type { Character } from "@/lib/character-types"

export const basicCharacter: Character = {
  id: "1",
  name: "Test",
  attributes: {
    fortitude: { base: 1, added: 0, bonus: 0 },
    finesse: { base: 1, added: 0, bonus: 0 },
    force: { base: 1, added: 0, bonus: 0 },
  },
  abilities: {
    athletics: { base: 0, added: 0, bonus: 0 },
    awareness: { base: 0, added: 0, bonus: 0 },
    closeCombat: { base: 2, added: 0, bonus: 0 },
    craft: { base: 0, added: 0, bonus: 0 },
    embassy: { base: 0, added: 0, bonus: 0 },
    integrity: { base: 0, added: 0, bonus: 0 },
    navigate: { base: 0, added: 0, bonus: 0 },
    physique: { base: 0, added: 0, bonus: 0 },
    presence: { base: 0, added: 0, bonus: 0 },
    performance: { base: 0, added: 0, bonus: 0 },
    rangedCombat: { base: 1, added: 0, bonus: 0 },
    sagacity: { base: 0, added: 0, bonus: 0 },
    stealth: { base: 0, added: 0, bonus: 0 },
    war: { base: 0, added: 0, bonus: 0 },
  },
  essence: { rating: 1, motes: 5, commitments: 0, spent: 0, anima: 0 },
  staticValues: {
    defenseModifier: 0,
    evasionModifier: 0,
    parryModifier: 0,
    resolveModifier: 0,
    soakModifier: 0,
    hardnessModifier: 0,
  },
  health: {
    baseline: { zero: 1, minusOne: 2, minusTwo: 2, incap: 1 },
    oxBodyLevels: 0,
    exaltType: "lunar",
    bashingDamage: 0,
    lethalDamage: 0,
    aggravatedDamage: 0,
    dramaticInjuries: [],
  },
  armor: [],
  weapons: [],
  milestones: { personal: 0, exalt: 0, minor: 0, major: 0 },
  advancement: [],
  dicePool: {
    attribute: "fortitude",
    ability: "athletics",
    targetNumber: 7,
    doublesThreshold: 10,
    extraDiceBonus: 0,
    extraDiceNonBonus: 0,
    extraSuccessBonus: 0,
    extraSuccessNonBonus: 0,
    isStunted: false,
  },
  charms: [],
  spells: [],
  combat: {
    power: 0,
    joinBattleDiceBonus: 0,
    joinBattleSuccessBonus: 0,
    decisiveExtraDice: 0,
    decisiveExtraSuccess: 0,
  },
  social: { virtues: { major: null, minor: null }, intimacies: [], backgrounds: [] },
  rulings: [],
}

