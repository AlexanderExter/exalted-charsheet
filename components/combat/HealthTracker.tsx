"use client";

import React from "react";
import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
import { Label } from "@/components/ui/label";
import { Input } from "@/components/ui/input";
import {
  Select,
  SelectContent,
  SelectItem,
  SelectTrigger,
  SelectValue,
} from "@/components/ui/select";
import {
  HealthLevelsSchema,
  HealthSchema,
  type Character,
  type ExaltType,
  type DramaticInjury,
} from "@/lib/character-types";
import type { CharacterCalculations } from "@/hooks/useCharacterCalculations";
import { DramaticInjuriesList } from "@/components/combat/DramaticInjuriesList";
import { z } from "zod";

interface HealthTrackerProps {
  character: Character;
  updateCharacter: (updates: Partial<Character>) => void;
  calculations: CharacterCalculations;
  getTotalHealthLevels: () => number;
  addDramaticInjury: () => void;
  updateDramaticInjury: <K extends keyof DramaticInjury>(
    id: string,
    field: K,
    value: DramaticInjury[K]
  ) => void;
  deleteDramaticInjury: (id: string) => void;
}

export const HealthTracker: React.FC<HealthTrackerProps> = ({
  character,
  updateCharacter,
  calculations,
  getTotalHealthLevels,
  addDramaticInjury,
  updateDramaticInjury,
  deleteDramaticInjury,
}) => {
  const [values, setValues] = React.useState({
    zero: String(character?.health?.baseline?.zero || 1),
    minusOne: String(character?.health?.baseline?.minusOne || 2),
    minusTwo: String(character?.health?.baseline?.minusTwo || 2),
    incap: String(character?.health?.baseline?.incap || 1),
    oxBodyLevels: String(character?.health?.oxBodyLevels || 0),
    bashingDamage: String(character?.health?.bashingDamage || 0),
    lethalDamage: String(character?.health?.lethalDamage || 0),
    aggravatedDamage: String(character?.health?.aggravatedDamage || 0),
  });

  const [errors, setErrors] = React.useState<Record<string, string>>({});

  React.useEffect(() => {
    setValues({
      zero: String(character?.health?.baseline?.zero || 1),
      minusOne: String(character?.health?.baseline?.minusOne || 2),
      minusTwo: String(character?.health?.baseline?.minusTwo || 2),
      incap: String(character?.health?.baseline?.incap || 1),
      oxBodyLevels: String(character?.health?.oxBodyLevels || 0),
      bashingDamage: String(character?.health?.bashingDamage || 0),
      lethalDamage: String(character?.health?.lethalDamage || 0),
      aggravatedDamage: String(character?.health?.aggravatedDamage || 0),
    });
  }, [character]);

  const baselineSchemas: Record<string, z.ZodTypeAny> = {
    zero: HealthLevelsSchema.shape.zero,
    minusOne: HealthLevelsSchema.shape.minusOne,
    minusTwo: HealthLevelsSchema.shape.minusTwo,
    incap: HealthLevelsSchema.shape.incap,
  };

  const handleBaselineChange =
    (field: keyof typeof baselineSchemas) => (e: React.ChangeEvent<HTMLInputElement>) => {
      const val = e.target.value;
      setValues(prev => ({ ...prev, [field]: val }));
      const result = baselineSchemas[field].safeParse(Number(val));
      if (result.success) {
        setErrors(prev => ({ ...prev, [field]: "" }));
        updateCharacter({
          health: {
            ...character.health,
            baseline: { ...character.health?.baseline, [field]: result.data },
          },
        });
      } else {
        setErrors(prev => ({
          ...prev,
          [field]: result.error.issues[0]?.message || "Invalid value",
        }));
      }
    };

  const handleOxBodyChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const val = e.target.value;
    setValues(prev => ({ ...prev, oxBodyLevels: val }));
    const result = HealthSchema.shape.oxBodyLevels.safeParse(Number(val));
    if (result.success) {
      setErrors(prev => ({ ...prev, oxBodyLevels: "" }));
      updateCharacter({
        health: { ...character.health, oxBodyLevels: result.data },
      });
    } else {
      setErrors(prev => ({
        ...prev,
        oxBodyLevels: result.error.issues[0]?.message || "Invalid value",
      }));
    }
  };

  const handleDamageChange =
    (field: "bashingDamage" | "lethalDamage" | "aggravatedDamage") =>
    (e: React.ChangeEvent<HTMLInputElement>) => {
      const val = e.target.value;
      setValues(prev => ({ ...prev, [field]: val }));
      const total = getTotalHealthLevels();
      const otherDamage =
        field === "bashingDamage"
          ? (character?.health?.lethalDamage || 0) + (character?.health?.aggravatedDamage || 0)
          : field === "lethalDamage"
            ? (character?.health?.bashingDamage || 0) + (character?.health?.aggravatedDamage || 0)
            : (character?.health?.bashingDamage || 0) + (character?.health?.lethalDamage || 0);
      const maxAllowed = Math.max(0, total - otherDamage);
      const schema = z.number().int().min(0).max(maxAllowed);
      const result = schema.safeParse(Number(val));
      if (result.success) {
        setErrors(prev => ({ ...prev, [field]: "" }));
        updateCharacter({
          health: { ...character.health, [field]: result.data },
        });
      } else {
        setErrors(prev => ({
          ...prev,
          [field]: result.error.issues[0]?.message || `Must be between 0 and ${maxAllowed}`,
        }));
      }
    };

  return (
    <Card>
      <CardHeader>
        <CardTitle className="flex items-center justify-between">
          Health Track
          <div className="flex items-center gap-4">
            <div className="flex items-center gap-2">
              <Label className="text-sm font-medium text-muted-foreground">Exalt Type:</Label>
              <Select
                value={character?.health?.exaltType || "lunar"}
                onValueChange={(value: ExaltType) =>
                  updateCharacter({
                    health: { ...character.health, exaltType: value },
                  })
                }
              >
                <SelectTrigger className="w-28">
                  <SelectValue />
                </SelectTrigger>
                <SelectContent>
                  <SelectItem value="lunar">Lunar</SelectItem>
                  <SelectItem value="solar">Solar</SelectItem>
                  <SelectItem value="dragon-blood">Dragon-Blood</SelectItem>
                  <SelectItem value="sidereal">Sidereal</SelectItem>
                  <SelectItem value="abyssal">Abyssal</SelectItem>
                  <SelectItem value="liminal">Liminal</SelectItem>
                  <SelectItem value="exigent">Exigent</SelectItem>
                </SelectContent>
              </Select>
            </div>
          </div>
        </CardTitle>
      </CardHeader>
      <CardContent>
        <div className="space-y-4">
          {/* Health Levels Grid */}
          <div className="grid grid-cols-4 gap-4">
            <div>
              <Label className="text-sm font-medium text-foreground/80">Zero Penalty</Label>
              <Input
                type="number"
                value={values.zero}
                onChange={handleBaselineChange("zero")}
                className="text-center"
                min={0}
              />
              {errors.zero && <p className="text-xs text-destructive mt-1">{errors.zero}</p>}
            </div>

            <div>
              <Label className="text-sm font-medium text-foreground/80">-1 Penalty</Label>
              <Input
                type="number"
                value={values.minusOne}
                onChange={handleBaselineChange("minusOne")}
                className="text-center"
                min={0}
              />
              {errors.minusOne && <p className="text-xs text-destructive mt-1">{errors.minusOne}</p>}
            </div>

            <div>
              <Label className="text-sm font-medium text-foreground/80">-2 Penalty</Label>
              <Input
                type="number"
                value={values.minusTwo}
                onChange={handleBaselineChange("minusTwo")}
                className="text-center"
                min={0}
              />
              {errors.minusTwo && <p className="text-xs text-destructive mt-1">{errors.minusTwo}</p>}
            </div>

            <div>
              <Label className="text-sm font-medium text-foreground/80">Incapacitated</Label>
              <Input
                type="number"
                value={values.incap}
                onChange={handleBaselineChange("incap")}
                className="text-center"
                min={0}
              />
              {errors.incap && <p className="text-xs text-destructive mt-1">{errors.incap}</p>}
            </div>
          </div>

          {/* Ox-Body Levels */}
          <div className="mt-4">
            <Label className="text-sm font-medium text-foreground/80">
              Ox-Body Levels (adds health levels)
            </Label>
            <Input
              type="number"
              value={values.oxBodyLevels}
              onChange={handleOxBodyChange}
              className="text-center"
              min={0}
            />
            {errors.oxBodyLevels && (
              <p className="text-xs text-destructive mt-1">{errors.oxBodyLevels}</p>
            )}
            <div className="mt-3 p-2 bg-muted/50 rounded">
              <div className="grid grid-cols-4 gap-2 text-xs">
                <div className="text-center">
                  <div className="font-medium text-success">0 Penalty</div>
                  <div className="text-lg font-bold">{calculations.healthLevels.zero}</div>
                </div>
                <div className="text-center">
                  <div className="font-medium text-warning">-1 Penalty</div>
                  <div className="text-lg font-bold">{calculations.healthLevels.minusOne}</div>
                </div>
                <div className="text-center">
                  <div className="font-medium text-warning">-2 Penalty</div>
                  <div className="text-lg font-bold">{calculations.healthLevels.minusTwo}</div>
                </div>
                <div className="text-center">
                  <div className="font-medium text-destructive">Incapacitated</div>
                  <div className="text-lg font-bold">{calculations.healthLevels.incap}</div>
                </div>
              </div>
            </div>

            {/* Incapacitation Rules */}
            {calculations.healthPenalty === -4 && (
              <div className="mt-3 border-t pt-3 bg-destructive/10 p-2 rounded">
                <div className="text-xs font-medium text-destructive mb-1">Incapacitation Rules:</div>
                <div className="text-xs text-destructive space-y-1">
                  <p>
                    Incapacitated characters have their Power reduced to 0 and cannot Build Power or
                    flurry. Allies can recover them by Building their Power to 10, which resets the
                    track to 0.
                  </p>
                  <p>
                    A player may ignore incapacitation by accepting a dramatic injury to an
                    Attribute or Primary Merit, applying a -1 die penalty or losing access to that
                    trait.
                  </p>
                  <p>
                    See
                    <a
                      href="https://exalted.fandom.com/wiki/Exalted_Essence"
                      target="_blank"
                      rel="noopener noreferrer"
                      className="underline ml-1"
                    >
                      Exalted Essence
                    </a>
                    for full rules.
                  </p>
                </div>
              </div>
            )}
          </div>

          {/* Damage Tracking */}
          <div className="grid grid-cols-3 gap-4">
            <div>
              <Label className="text-sm font-medium text-foreground/80">Bashing Damage</Label>
              <Input
                type="number"
                value={values.bashingDamage}
                onChange={handleDamageChange("bashingDamage")}
                className="text-center"
                min={0}
                max={Math.max(
                  0,
                  getTotalHealthLevels() -
                    ((character?.health?.lethalDamage || 0) +
                      (character?.health?.aggravatedDamage || 0))
                )}
              />
              {errors.bashingDamage && (
                <p className="text-xs text-destructive mt-1">{errors.bashingDamage}</p>
              )}
            </div>

            <div>
              <Label className="text-sm font-medium text-foreground/80">Lethal Damage</Label>
              <Input
                type="number"
                value={values.lethalDamage}
                onChange={handleDamageChange("lethalDamage")}
                className="text-center"
                min={0}
                max={Math.max(
                  0,
                  getTotalHealthLevels() -
                    ((character?.health?.bashingDamage || 0) +
                      (character?.health?.aggravatedDamage || 0))
                )}
              />
              {errors.lethalDamage && (
                <p className="text-xs text-destructive mt-1">{errors.lethalDamage}</p>
              )}
            </div>

            <div>
              <Label className="text-sm font-medium text-foreground/80">Aggravated Damage</Label>
              <Input
                type="number"
                value={values.aggravatedDamage}
                onChange={handleDamageChange("aggravatedDamage")}
                className="text-center"
                min={0}
                max={Math.max(
                  0,
                  getTotalHealthLevels() -
                    ((character?.health?.bashingDamage || 0) +
                      (character?.health?.lethalDamage || 0))
                )}
              />
              {errors.aggravatedDamage && (
                <p className="text-xs text-destructive mt-1">{errors.aggravatedDamage}</p>
              )}
            </div>
          </div>

          {/* Dramatic Injuries */}
          <DramaticInjuriesList
            injuries={character.health?.dramaticInjuries || []}
            addDramaticInjury={addDramaticInjury}
            updateDramaticInjury={updateDramaticInjury}
            deleteDramaticInjury={deleteDramaticInjury}
          />
        </div>
      </CardContent>
    </Card>
  );
};

HealthTracker.displayName = "HealthTracker";
