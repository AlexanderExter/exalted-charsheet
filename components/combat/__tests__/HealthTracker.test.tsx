import "@testing-library/jest-dom";
import React from "react";
import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { HealthTracker } from "@/components/combat/HealthTracker";
import { createDefaultHealth } from "@/lib/character-defaults";
import type { Character } from "@/lib/character-types";
import type { CharacterCalculations } from "@/hooks/useCharacterCalculations";

describe("HealthTracker", () => {
  it("renders incapacitation rules snapshot", () => {
    const character = { health: createDefaultHealth() } as unknown as Character;
    const calculations = {
      healthPenalty: -4,
      healthLevels: { zero: 2, minusOne: 2, minusTwo: 2, incap: 1 },
    } as unknown as CharacterCalculations;

    const { container } = render(
      <HealthTracker
        character={character}
        updateCharacter={() => {}}
        calculations={calculations}
        getTotalHealthLevels={() => 7}
        addDramaticInjury={() => {}}
        updateDramaticInjury={() => {}}
        deleteDramaticInjury={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });
});

