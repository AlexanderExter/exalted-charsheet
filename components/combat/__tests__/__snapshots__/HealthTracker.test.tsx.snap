// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HealthTracker > renders incapacitation rules snapshot 1`] = `
<div>
  <div
    class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm"
    data-slot="card"
  >
    <div
      class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6"
      data-slot="card-header"
    >
      <div
        class="leading-none font-semibold flex items-center justify-between"
        data-slot="card-title"
      >
        Health Track
        <div
          class="flex items-center gap-4"
        >
          <div
            class="flex items-center gap-2"
          >
            <label
              class="flex items-center gap-2 select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-sm font-medium text-gray-600"
              data-slot="label"
            >
              Exalt Type:
            </label>
            <button
              aria-autocomplete="none"
              aria-controls="radix-«r0»"
              aria-expanded="false"
              class="border-input data-[placeholder]:text-muted-foreground [&_svg:not([class*='text-'])]:text-muted-foreground focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive dark:bg-input/30 dark:hover:bg-input/50 flex items-center justify-between gap-2 rounded-md border bg-transparent px-3 py-2 text-sm whitespace-nowrap shadow-xs transition-[color,box-shadow] outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 data-[size=default]:h-9 data-[size=sm]:h-8 *:data-[slot=select-value]:line-clamp-1 *:data-[slot=select-value]:flex *:data-[slot=select-value]:items-center *:data-[slot=select-value]:gap-2 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 w-28"
              data-size="default"
              data-slot="select-trigger"
              data-state="closed"
              dir="ltr"
              role="combobox"
              type="button"
            >
              <span
                data-slot="select-value"
                style="pointer-events: none;"
              >
                Lunar
              </span>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down size-4 opacity-50"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="px-6"
      data-slot="card-content"
    >
      <div
        class="space-y-4"
      >
        <div
          class="grid grid-cols-2 md:grid-cols-4 gap-4"
        >
          <div>
            <label
              class="flex items-center gap-2 select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-sm font-medium text-gray-700"
              data-slot="label"
            >
              Zero Penalty
            </label>
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-center"
              data-slot="input"
              min="0"
              type="number"
              value="2"
            />
          </div>
          <div>
            <label
              class="flex items-center gap-2 select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-sm font-medium text-gray-700"
              data-slot="label"
            >
              -1 Penalty
            </label>
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-center"
              data-slot="input"
              min="0"
              type="number"
              value="2"
            />
          </div>
          <div>
            <label
              class="flex items-center gap-2 select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-sm font-medium text-gray-700"
              data-slot="label"
            >
              -2 Penalty
            </label>
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-center"
              data-slot="input"
              min="0"
              type="number"
              value="2"
            />
          </div>
          <div>
            <label
              class="flex items-center gap-2 select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-sm font-medium text-gray-700"
              data-slot="label"
            >
              Incapacitated
            </label>
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-center"
              data-slot="input"
              min="0"
              type="number"
              value="1"
            />
          </div>
        </div>
        <div
          class="mt-4"
        >
          <label
            class="flex items-center gap-2 select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-sm font-medium text-gray-700"
            data-slot="label"
          >
            Ox-Body Levels (adds health levels)
          </label>
          <input
            class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-center"
            data-slot="input"
            min="0"
            type="number"
            value="0"
          />
          <div
            class="mt-3 p-2 bg-gray-50 rounded"
          >
            <div
              class="grid grid-cols-4 gap-2 text-xs"
            >
              <div
                class="text-center"
              >
                <div
                  class="font-medium text-green-600"
                >
                  0 Penalty
                </div>
                <div
                  class="text-lg font-bold"
                >
                  2
                </div>
              </div>
              <div
                class="text-center"
              >
                <div
                  class="font-medium text-yellow-600"
                >
                  -1 Penalty
                </div>
                <div
                  class="text-lg font-bold"
                >
                  2
                </div>
              </div>
              <div
                class="text-center"
              >
                <div
                  class="font-medium text-orange-600"
                >
                  -2 Penalty
                </div>
                <div
                  class="text-lg font-bold"
                >
                  2
                </div>
              </div>
              <div
                class="text-center"
              >
                <div
                  class="font-medium text-red-600"
                >
                  Incapacitated
                </div>
                <div
                  class="text-lg font-bold"
                >
                  1
                </div>
              </div>
            </div>
          </div>
          <div
            class="mt-3 border-t pt-3 bg-red-50 p-2 rounded"
          >
            <div
              class="text-xs font-medium text-red-700 mb-1"
            >
              Incapacitation Rules:
            </div>
            <div
              class="text-xs text-red-600 space-y-1"
            >
              <p>
                Incapacitated characters have their Power reduced to 0 and cannot Build Power or flurry. Allies can recover them by Building their Power to 10, which resets the track to 0.
              </p>
              <p>
                A player may ignore incapacitation by accepting a dramatic injury to an Attribute or Primary Merit, applying a -1 die penalty or losing access to that trait.
              </p>
              <p>
                See
                <a
                  class="underline ml-1"
                  href="https://exalted.fandom.com/wiki/Exalted_Essence"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Exalted Essence
                </a>
                for full rules.
              </p>
            </div>
          </div>
        </div>
        <div
          class="grid grid-cols-3 gap-4"
        >
          <div>
            <label
              class="flex items-center gap-2 select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-sm font-medium text-gray-700"
              data-slot="label"
            >
              Bashing Damage
            </label>
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-center"
              data-slot="input"
              max="7"
              min="0"
              type="number"
              value="0"
            />
          </div>
          <div>
            <label
              class="flex items-center gap-2 select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-sm font-medium text-gray-700"
              data-slot="label"
            >
              Lethal Damage
            </label>
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-center"
              data-slot="input"
              max="7"
              min="0"
              type="number"
              value="0"
            />
          </div>
          <div>
            <label
              class="flex items-center gap-2 select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-sm font-medium text-gray-700"
              data-slot="label"
            >
              Aggravated Damage
            </label>
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-center"
              data-slot="input"
              max="7"
              min="0"
              type="number"
              value="0"
            />
          </div>
        </div>
        <div
          class="space-y-3"
        >
          <div
            class="flex items-center justify-between"
          >
            <label
              class="flex items-center gap-2 select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-sm font-medium text-gray-700"
              data-slot="label"
            >
              Dramatic Injuries
            </label>
            <button
              class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5"
              data-slot="button"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-plus w-4 h-4 mr-1"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 12h14"
                />
                <path
                  d="M12 5v14"
                />
              </svg>
              Add Injury
            </button>
          </div>
          <p
            class="text-gray-500 italic text-sm"
          >
            No dramatic injuries.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
